### PR TITLE
chore: update changelog for v2.3.0

### PR DESCRIPTION
**Pull request by dbolduc**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9963)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -116,7 +116,9 @@ may elicit a deprecation warning. See [#9086] for more details.
 
 </details>
 
-## v2.3.0 - TBD
+## v2.4.0 - TBD
```
Comment (devbww) [inline CHANGELOG.md:119]: I think the new convention, which I like, is not to add this until the "bump" stage.

File: CHANGELOG.md [hunk 2]
```diff
@@ -126,6 +126,32 @@ the APIs in these libraries are stable, and are ready for production use.
...
+### [BigQuery](/google/cloud/bigquery/README.md)
+
+The library has been expanded to include [Analytics Hub][bq-analytics-hub], an
+API that facilitates data sharing within and across organizations.
+([#9882](https://github.com/googleapis/google-cloud-cpp/pull/9882))
+
+### [Spanner](/google/cloud/spanner/README.md)
+
+- fix(spanner): remove session from pool upon "not found" refresh failure ([#9954](https://github.com/googleapis/google-cloud-cpp/pull/9954))
+- feat(spanner): add support for CMMR phase 2 ([#9872](https://github.com/googleapis/google-cloud-cpp/pull/9872))
```
Comment (coryan) [inline CHANGELOG.md:138]: Maybe expand what CMMR means?

**Review by coryan — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (dbolduc) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (dbolduc) [reply CHANGELOG.md]: Done.

**Pull request merged**